google/cloud/storage/_helpers.py

```diff
@@ -190,7 +190,9 @@ def reload(
                                             blob's current metageneration does not match the given value.
 
         :type retry: google.api_core.retry.Retry
```

Comment:
Technically either google.api_core.retry.Retry or google.cloud.storage.retry.ConditionalRetryPolicy are supported here, and in some cases ConditionalRetryPolicy is the default so despite the complexity we should document it here.